src/test/java/oop/animals/service/EnclosureServiceTest.java

```diff
@@ -0,0 +1,167 @@
+    }
+
+    @Test
+    public void createEnclosureTest() {
+        enclosureService.createEnclosure();
+        assertEquals(1, enclosureService.getEnclosuresSize());
+    }
+
+    @Test
+    public void createEnclosureAndAddAnimalTest() {
+        Lion lion = new Lion();
+        lion.setAge(10);
+        lion.setWeight(35);
+        lion.setWeaponType(Predator.WeaponType.TEETH);
+
+        enclosureService.createEnclosureAndAddAnimal(lion);
+        final Enclosure<Mammal> specificEnclosure = enclosureService.getEnclosures().get(0);
+
+        assertTrue(specificEnclosure.containsOf(lion));
+        assertEquals(1, enclosureService.getEnclosuresSize());
+        assertEquals(1, specificEnclosure.getAnimals().size());
+    }
+
+    @Test
+    public void addAnimalToAnySuitableEnclosure() {
+        Lion lion = new Lion();
+        lion.setAge(10);
+        lion.setWeight(35);
+        lion.setWeaponType(Predator.WeaponType.TEETH);
+
+        Coyote coyote = new Coyote();
+        coyote.setAge(3);
+        coyote.setWeight(10);
+        coyote.setWeaponType(Predator.WeaponType.CLAWS);
+
+        Cachalot cachalot = new Cachalot();
+        cachalot.setAge(15);
+        cachalot.setWeight(500);
+        cachalot.setWeaponType(Predator.WeaponType.TEETH);
+
+        KillerWhale killerWhale = new KillerWhale();
+        killerWhale.setAge(9);
+        killerWhale.setWeight(90);
+        killerWhale.setWeaponType(Predator.WeaponType.TEETH);
+
+        JungleCat jungleCat = new JungleCat();
+        jungleCat.setAge(5);
+        jungleCat.setWeight(48);
+        jungleCat.setWeaponType(Predator.WeaponType.CLAWS);
+
+        WhiteBear whiteBear = new WhiteBear();
+        whiteBear.setAge(20);
+        whiteBear.setWeight(160);
+        whiteBear.setWeaponType(Predator.WeaponType.CLAWS);
+
+        List<Mammal> animalsOne = new ArrayList<>();
+        animalsOne.add(cachalot);
+        animalsOne.add(killerWhale);
+        animalsOne.add(jungleCat);
+        animalsOne.add(whiteBear);
+
+        List<Mammal> animalsTwo = new ArrayList<>();
+        animalsTwo.add(coyote);
+        animalsTwo.add(cachalot);
+        animalsTwo.add(killerWhale);
+        animalsTwo.add(jungleCat);
+        animalsTwo.add(whiteBear);
+
+        enclosureService.createEnclosureAndAddAnimals(animalsTwo);
+        enclosureService.createEnclosureAndAddAnimals(animalsOne);
+
+        enclosureService.addAnimalToAnySuitableEnclosure(lion);
+        final Enclosure<Mammal> specificEnclosure = enclosureService.getEnclosures().get(1);
+
+        assertTrue(specificEnclosure.containsOf(lion));
+        assertEquals(2, enclosureService.getEnclosuresSize());
+        assertEquals(5, specificEnclosure.getAnimals().size());
+    }
+
+    @Test
+    public void addAnimalToSpecificEnclosureTest() {
+        Lion lion = new Lion();
+        lion.setAge(10);
+        lion.setWeight(35);
+        lion.setWeaponType(Predator.WeaponType.TEETH);
```

Comment:
Please use constructor with parameters here after you update your model classes.